addons/web/static/src/js/views/control_panel/control_panel_model.js

```diff
@@ -435,18 +436,46 @@ var ControlPanelModel = mvc.Model.extend({
      */
     toggleFilterWithOptions: function (filterId, optionId) {
         var filter = this.filters[filterId];
+        optionId = optionId || filter.defaultOptionId;
         var group = this.groups[filter.groupId];
-        var alreadyActive = group.activeFilterIds.indexOf(filterId) !== -1;
-        if (alreadyActive) {
-            if (filter.currentOptionId === optionId) {
+        if (filter.type === 'filter') {
+            var alreadyActive = group.activeFilterIds.findIndex(isIso([filterId])) !== -1;
+            if (alreadyActive) {
```

Comment:
can you comment those ifs so we know the role of each case

Reply:
ok